src/main/java/core/operations/BalanceOperation.java

```
@@ -0,0 +1,15 @@
+package core.operations;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.StorageDao;
+import core.interfaces.OperationHandler;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction tx, StorageDao storageDao) {
+        if (tx != null && tx.getQuantity() >= 0) {

```

The handler currently silently ignores invalid input (it simply does nothing if `tx` is null or quantity < 0). The checklist requires proper error handling and informative RuntimeExceptions: "Don't ignore exceptions in try-catch blocks ... Let's rethrow a RuntimeException with an **informative** message and exception object." Instead of swallowing/ignoring, validate and throw a RuntimeException with details about the bad transaction (line content or transaction object).